File: trivial_graphs.cpp
// Basic

#include <iostream>
using namespace std;

class Graph{
    int V;
    int** adj;

public:
    Graph( int vertices ){
        V = vertices;
        adj = new int*[V];
        for( int i = 0 ; i < V ; i++ ){
            adj[i] = new int[V]{ 0 };
        }
    }

    void addEdge( int u, int v ){
        adj[u][v] = 1;
        adj[v][u] = 1;
    }

    void display(){
        for( int i = 0 ; i < V ; i++ ){
            for( int j = 0 ; j < V ; j++ ){
                cout << adj[i][j] << " ";
            }
            cout << endl;
        }
    }

    ~Graph(){
        for( int i = 0 ; i < V ; i++ ) delete[] adj[i];
        delete[] adj;
    }
};

int main(){
    Graph trivial( 1 );
    trivial.display();

    Graph g( 3 );
    g.addEdge( 0, 1 );
    g.addEdge( 1, 2 );
    g.display();

    return 0;
}

File: trivial_graphs.java
// Basic

import java.util.Arrays;

public class Graph{
    private int[][] matrix;

    public Graph( int vertices ){
        matrix = new int[vertices][vertices];
    }

    public void addEdge( int u, int v ){
        matrix[u][v] = 1;
        matrix[v][u] = 1;
    }

    public void display(){
        for( int[] row : matrix ){
            System.out.println( Arrays.toString( row ) );
        }
    }

    public static void main( String[] args ){
        Graph trivial = new Graph( 1 );
        trivial.display();

        Graph g = new Graph( 3 );
        g.addEdge( 0, 1 );
        g.addEdge( 1, 2 );
        g.display();
    }
}
